### PR TITLE
Wr poetry

### PR DESCRIPTION
**Pull request by aschroed**
Changed packaging to poetry
- scripts and notebooks moved into a src directory so can be installed as a package
   not sure this is the best way to handle this as it required changing a ton of imports and mocker calls for tests etc. 
   however I think this allows the scripts and notebooks to potentially go to pypi and be pip installed

There is currently no dependency on invoke or jupyter (so while testing I installed via pip into the virtualenv) - not sure if they should be added or not?

Basically a learning exercise for me so sorry if this doesn't look right.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+homepage = "https://github.com/4dn-dcic/dcicwrangling"
+repository = "https://github.com/4dn-dcic/dcicwrangling"
+
+packages = [
+  { include="src", from="." }
+]
+
+[tool.poetry.dependencies]
+python = ">=3.6, <3.8"
+dcicutils = ">=0.31.1"
```
Comment (willronchetti) [inline pyproject.toml]: I would recommend a caret requirement here rather than `>=`, or if you're going to use that at least add `<1` as well.

File: tests/test_dcicwrangling.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from src import __version__
```
Comment (willronchetti) [inline tests/test_dcicwrangling.py]: I think this file can be removed altogether.

File: src/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+__version__ = '0.1.0'
```
Comment (willronchetti) [inline src/__init__.py]: You do not need this anymore with Poetry.

**Review by willronchetti — COMMENTED** [review]
Generally looks reasonable to me

File: .travis.yml [hunk 4]
```diff
@@ -3,29 +3,22 @@ sudo: false
...
-  password:
-    secure: dP62awneMbu7jlpFmkQpwGqDxmqlOJorbWjh51BLOuPDc/LHOlI50vdLCwNLwfQFedKoP5LZIfWYBoy8wbXLhXpqZkgwwPvBesaUPuRONdxLqXCqM21d/1nmgIEIdwgFkcIKTRwTPK5LKygXg5iAstkrq+kcdv0Hwake4QE7ZtJEo2kTKPF8jkzHteeBfSB1n6GrQ6iQnkzMg6RiqjB3t0sdNRX/sKZqM5CdYzA523tDjrPTGTWyhVS++/TeNW5PZajZyVr2HKH5+OYp5Ik/8w0Q6EXM51ZuFrqTnv67kTCSIfXai6tpCauI3x3EiUVhkyznIXz2uC5i518GWQWwbuf+k4IKmhXtUe6uA+rVaCkdBOjBN6CyLGJwKaefCJw/RMgWKnqqrHr+6hfwKJctQVxDQcXpvP3+O+LkqBY3D1CYtSjAiini3ooKa2UQGGBhCe8HcPXsMwzPzefdv36RdGASg4X27qOTCKm4F+Jzo9LWF5K3AmYDyGUOXlpfSK1gug6DBrp3VsE3rNX3z8PRj5+LZhDXKg/K99FzM3lsoYypP/V5btoZ3rbUvklRueotSjIO2mohzX4aJ7cZt8hMuoABaXdyIEggWccPYCkOnBnJ0yiXp/PpiRdXeVF2G7eo5fRM/TWPAkvNUJpFvQBRCtvW3xmirY+7Rf79S8FntHM=
-  on:
-    tags: true
-  distributions: sdist bdist_wheel
-install:
-- pip install -U -r requirements-dev.txt
-- pip install -U -r requirements.txt
+- '3.6'
```
Comment (willronchetti) [inline .travis.yml:6]: `pyproject.toml` does not rule out 3.7, so you should probably have a 3.7 build as well (if you think it is necessary). If this is a mistake disallow 3.7 in `pyproject.toml`.

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+description = "Scripts and Jupyter notebooks for 4DN wrangling"
+authors = ["4DN-DCIC Team <support@4dnucleome.org>"]
+license = "MIT"
+readme = "README.md"
+homepage = "https://github.com/4dn-dcic/dcicwrangling"
+repository = "https://github.com/4dn-dcic/dcicwrangling"
+packages = [{ include="scripts", from="." }]
+
+[tool.poetry.dependencies]
+python = ">=3.6, <3.8"
```
Comment (willronchetti) [inline pyproject.toml:13]: As stated previously, if you don't support 3.7 just explicitly disallow it here. I'm guessing it is not much work to be compatible, but not sure.

**Review by willronchetti — APPROVED** [review]
Looks good to me!
